src/liballoc/collections/btree/map.rs

```diff
@@ -1634,9 +1634,11 @@ impl<'a, K, V> RangeMut<'a, K, V> {
 
         let mut cur_handle = match handle.right_kv() {
             Ok(kv) => {
-                let (k, v) = ptr::read(&kv).into_kv_mut();
-                self.front = kv.right_edge();
-                return (k, v);
+                self.front = ptr::read(&kv).right_edge();
+                // Doing the descend invalidates the references returned by `into_kv_mut`,
+                // so we have to do this last.
```

Comment:
I struggle with this comment because there's no descend here, right? Maybe the right_edge also invalidates references, or the change is just here to match the code below where a descend happens.

Reply:
Well, there's still a call to `right_edge` so I wanted to be sure that wouldn't cause any problems, without actually checking in detail what that function does. The comment is probably a copy-paste mistake though.

Reply:
I've taken the plunge and collapsed the two copies in each iterator [in my #66747 PR](https://github.com/rust-lang/rust/pull/66747/files#diff-02bad9328a699a220dca897c0457c1dc)